nautobot/extras/migrations/0099_remove_dangling_note_objects.py

```diff
@@ -0,0 +1,27 @@
+# Generated by Django 3.2.22 on 2023-11-06 22:56
+
+from django.db import migrations
+
+
+def remove_dangling_note_objects(apps, schema_editor):
+    """
+    Remove Note objects associated with the already deleted object.
+    """
+    Note = apps.get_model("extras", "Note")
+    print("\n>>> Removing invalid and dangling Note objects ...\n")
+    for note in Note.objects.all():
+        model_class = apps.get_model(note.assigned_object_type.app_label, note.assigned_object_type.model)
+        try:
+            model_class.objects.get(id=note.assigned_object_id)
+        except model_class.DoesNotExist:
+            note.delete()
```

Comment:
I know this is a little complicated but `Note.objects.filter(assigned_object=None).delete()` does not work since `assigned_object` is a GFK field